### ISSUE TITLE
More related tools (and another one named mdsh)

### ISSUE THREAD

**Issue by pjeby** [zimbatm/mdsh#25]
Hi there!  I stumbled across your project while searching for something related to [this other `mdsh` project](https://github.com/bashup/mdsh), and just thought I should mention the possible naming conflict.  The mdsh I wrote is a tool for doing both data processing and execution of blocks within markdown, implemented as a shell script.  It has third-party extensions such [mdshi](https://github.com/Feeld/mdshi), which is actually kind of close to some of what your mdsh is for.

Anyway, I can't really lay claim to the name even if I used it first, but it might be nice if you named yours something more about its function (md-live? mdexec? mdgen?) and less about its shellness, since yours isn't actually written *in* shell and so the shell aspect of it is kind of incidental.  Whereas mine is written *in* shell and compiles markdown into shell scripts, so the shellness is part of its essential nature.  ;-)

Also, as an FYI, there's a tool you might be interested in called [cram](https://bitheap.org/cram/) that does something very similar to your mdsh as well, but it's for testing anything that can be tested by comparing a file's content with the output of shell commands, .  It processes indented blocks like this:

    $ ls
    foo
    bar

...by running the commands that begin with `$ ` (optionally continued with `> `), and verifying the output matches the input.  If you set its indentation width to 4, then it can process markdown indented code blocks, and it can also be used to generate the data in the first place, because if the output doesn't match the input file, you can tell it to update the input file with the new output.

I actually use cram for testing mdsh (my mdsh, that is) and other shell tools, as it's nice to be able to have the documentation examples tested, as well as being able to document the tests.  :-)

So, these all may or may not be projects you might or might not want to mention in the "Related Projects" part of your readme.  Hope you find some or any of this illuminating and/or useful.

**Comment by zimbatm** [comment]
Hey @pjeby, thanks for letting me know about these other projects, it's always cool to discover new things and the approach other people took.

My `mdsh` is called like that because it's markdown + executable shell blocks. It seems a bit simpler in that it doesn't allow to run arbitrary languages, only bash. One goal was to get a sort of jupyter notebook bash kernel, but with a readable serialization format.

I will try and come up with a better name for my project but not guarantee. For now I haven't really found another name that I like.

**Comment by pjeby** [comment]
No problem.  Like I said, just thought you should know.  :smile:

**Comment by zimbatm** [comment]
This topic came back a couple of times in my brain but I still haven't found a better name yet.

**Comment by pjeby** [comment]
Huh.  Small world!  Anyway, don't worry about it, it's really not a big deal.

**Comment by elviejo79** [comment]
As an external user. I just want to mention that it is confusing to have two projects, with similar functionality named the same... While googling I would find documentation to both and thought they were the same thing!

**Comment by airtonix** [comment]
Yep, AI gets confused now, so people are more likely to not use either of your projects because they try using syntax from one in the other.
